packages/driver-interface/src/botRoutes.ts

```diff
@@ -63,7 +63,17 @@ export const init = (bot: Telegraf<TelegrafContext>): void => {
           instructionGroupId
         )
       case 'delivered':
-        cache.setDriverDoneDelivering(telegramId)
+        return cache
+          .setDriverDoneDelivering(telegramId)
+          .then(() =>
+            Promise.all(
+              bookingIds.map((bookingId: string) =>
+                amqp.publishBookingEvent(bookingId, 'delivered')
```

Comment:
Is there a need for awaiting this in order to continue?

Reply:
We can probably remove the promise all, yes.

Reply:
Bot on the other hand, I'd rather have predictible behaviour, with this chain, just glancing over it, you'd expect all events to be published before the next "then" is triggered...So I'll let it stay.